### PR TITLE
Fix l10n dir

### PR DESCRIPTION
**Pull request by sgn**
See  #27491

### CODE REVIEW

**Comment by q66** [comment]
let's not put `--with-locales-dir` into default `gnu-configure-args` since that's xfce specific

also let's properly patch the xfce m4 files as well

File: common/hooks/post-install/14-pkglint-locale-in-usr-lib.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# vim: set ts=4 sw=4 et:
+#
+# This hook executes the following tasks:
+#   - Looks on all packages for l10n files being installed to /usr/share
+
+hook() {
+    if [ -d $PKGDESTDIR/usr/lib/locale ]; then
+        msg_error "${pkgver}: l10n files found in /usr/lib\n"
```
Comment (Chocimier) [inline common/hooks/post-install/14-pkglint-locale-in-usr-lib.sh:8]: Message would be more helpful with tip how to proceed, eg. _l10n files found in /usr/lib; use /usr/share_

Comment (ericonr) [reply common/hooks/post-install/14-pkglint-locale-in-usr-lib.sh:8]: Maybe mention `locale` explicitly? So `/usr/lib/locale` and `/usr/share/locale`.

Comment (q66) [reply common/hooks/post-install/14-pkglint-locale-in-usr-lib.sh:8]: just put it in the same hook that checks other banned directories...

**Comment by q66** [comment]
what's up with that github workflows commit? also did you patch the xfce4-dev-tools m4 file? I don't see that in the commits

**Comment by sgn** [comment]
> what's up with that github workflows commit?

I would like to turn it off to let the build finish, I'll remove it before pushing.

> also did you patch the xfce4-dev-tools m4 file? I don't see that in the commits.

I supposed we can just add `--with-locales-dir=...` into all xfce4 packages instead of patching its m4 file.

**Comment by q66** [comment]
as I said, we still need to patch the file, so that when people build things on their system without xbps-src it still builds correctly

it's a bug and still needs to be fixed either way

**Comment by q66** [comment]
also, I don't really like dumping everything into one big PR

things that are uncontroversial and not blocked on anything can be pushed separately (this applies to the xfce templates needing --with-locales-dir), I have some potential objections to how the m4 files are modified, but I need to look deeper into that first

**Comment by sgn** [comment]
I'll push xfce4 stuff out, now.

**Comment by sgn** [comment]
I think we should push the hook out now?

**Comment by q66** [comment]
yeah, i guess let's push the hook

**Comment by ericonr** [comment]
This will break people trying to build stuff on musl in the meantime...

The hook shouldn't error out now, it should only warn.

**Comment by q66** [comment]
it's fine for it to error, i doubt anyone will run into it and we can fix it afterwards

**Comment by sgn** [comment]
```
=> libmcrypt-2.5.8_3: running do_configure ...
configure: error: unrecognized option: --localedir=/usr/share/locale
```

**Comment by q66** [comment]
let's just drop localedir from the defaults, since according to autoconf manual it already defaults to what we want either way, it's just glib and xfce that do things wrong

**Comment by sgn** [comment]
> let's just drop localedir from the defaults, since according to autoconf manual it already defaults to what we want either way, it's just glib and xfce that do things wrong

I agree. Let's see if I could re-work on this weekend.

**Comment by sgn** [comment]
Tested:
- [ ] GConf
- [x] ario
- [x] appdata-tools
- [x] brasero
- [x] dia
- [x] desmume
- [x] gdmap
- [x] gconf-editor
- [ ] gimp-lqr-plugin
- [x] gmpc
- [x] geeqie
- [x] gnome-system-log
- [x] gnome-mime-data
- [x] gnumeric
- [x] gnome-mplayer
- [x] gnome-nettool
- [ ] gnome-doc-utils
- [x] gpicview
- [ ] gnome-screensaver
- [ ] gtk+
- [x] gtkspell
- [x] gxmessage
- [ ] gtk2-engines
- [ ] gtkpod
- [ ] gtksourceview2
- [ ] homebank
- [ ] leafpad
- [ ] libcryptui
- [ ] libgpod
- [ ] libglib-static
- [ ] libgnome-keyring
- [ ] libgnomecanvas
- [ ] libwnck2
- [ ] lightdm-gtk-greeter
- [ ] lxappearance-obconf
- [ ] lxinput
- [ ] lxlauncher
- [ ] pcmanx-gtk2
- [ ] parcellite
- [ ] pmount
- [ ] pidgin-libnotify
- [ ] pidgin-otr
- [ ] pidgin-privacy-please
- [ ] polkit-gnome
- [ ] pragha
- [ ] seahorse-nautilus
- [ ] seahorse-sharing
- [ ] spacefm
- [ ] tilp2
- [ ] udevil
- [ ] udisks
- [ ] ufraw
- [ ] uget
- [ ] vte
- [ ] vte290
- [ ] viking
- [ ] volumeicon
- [ ] xdg-user-dirs-gtk
- [ ] xsane
- [ ] xnoise
- [ ] xournal

**Comment by sgn** [comment]
The last batch. I'm not touching glib and xfce-dev-tools since q66 had expressed intends to fix them.

**Pull request merged**